### PR TITLE
store: introduce cold storage

### PR DESCRIPTION
**Pull request by mina86**
Add a `cold_store` Cargo feature which enables the option to configure
the node with cold storage.  At the moment, all this does is open the
cold database and doesn’t enable any other features.  The idea is that
this can now allow experimenting with code that needs access to the
cold storage.

### CODE REVIEW

**Comment by akhi3030** [comment]
@posvyatokum: I'm happy to approve once your reviews are in and you approve.

File: core/store/src/opener.rs
Comment (posvyatokum) [inline core/store/src/opener.rs:30]: Wrong comment (copy from previous error)

File: core/store/src/opener.rs
Comment (posvyatokum) [inline core/store/src/opener.rs:37]: Same here, wrong comment (copy from HotColdExistenceMismatch)

File: nearcore/src/lib.rs
Comment (posvyatokum) [inline nearcore/src/lib.rs:78]: but -> both?

**Review by posvyatokum — APPROVED** [review]
Code looks good, all my comments are about comments.

**Review by akhi3030 — APPROVED** [review]

**Pull request merged**
